aws/resource_aws_cloudwatch_log_destination.go

```diff
@@ -64,28 +62,34 @@ func resourceAwsCloudWatchLogDestinationPut(d *schema.ResourceData, meta interfa
 		TargetArn:       aws.String(target_arn),
 	}
 
-	return resource.Retry(3*time.Minute, func() *resource.RetryError {
-		resp, err := conn.PutDestination(params)
+	var resp *cloudwatchlogs.PutDestinationOutput
+	var err error
+	err = resource.Retry(3*time.Minute, func() *resource.RetryError {
+		resp, err = conn.PutDestination(params)
 
-		if err == nil {
-			d.SetId(name)
-			d.Set("arn", *resp.Destination.Arn)
-		}
-
-		awsErr, ok := err.(awserr.Error)
-		if !ok {
+		if isAWSErr(err, cloudwatchlogs.ErrCodeInvalidParameterException, "Could not deliver test message to specified") {
 			return resource.RetryableError(err)
 		}
-
-		if awsErr.Code() == "InvalidParameterException" {
-			if strings.Contains(awsErr.Message(), "Could not deliver test message to specified") {
-				return resource.RetryableError(err)
-			}
+		if isAWSErr(err, cloudwatchlogs.ErrCodeInvalidParameterException, "") {
```

Comment:
In this case, we should update the rest of this logic to just:

```go
if err != nil {
	return resource.NonRetryableError(err)
}

return nil
```

Removing the `isAWSErr(err, cloudwatchlogs.ErrCodeInvalidParameterException, "")` and `isAWSErr(err, "", "")` conditions.

The handling of `isAWSErr(err, cloudwatchlogs.ErrCodeInvalidParameterException, "")` would be caught by the above and `isAWSErr(err, "", "")` is invalid as all AWS Go SDK errors should have an error code.